lib/model/content.dart

```diff
@@ -990,14 +990,14 @@ class _ZulipContentParser {
       if (child is! dom.Element) return null;
       if (child.localName != 'pre') return null;
 
-      if (child.nodes.length > 2) return null;
+      if (child.nodes.length > 2 || child.nodes.isEmpty) return null;
       if (child.nodes.length == 2) {
         final first = child.nodes[0];
         if (first is! dom.Element
             || first.localName != 'span'
             || first.nodes.isNotEmpty) return null;
       }
-      final grandchild = child.nodes[child.nodes.length - 1];
+      final grandchild = child.nodes.last;
```

Comment:
Is this a pure refactor, or does this change behavior somehow? I noticed there's a line `final first = child.nodes[0];` above this; should that also be changed to say `child.nodes.first`?

Reply:
This refactoring is purely for readability. Let's get an nfc commit for both.